network-grpc/src/client.rs

```diff
@@ -385,13 +385,19 @@ where
 
 impl<T, S, E> HeaderService<T> for Client<S, E>
 where
-    T: HasHeader,
+    T: Block + HasHeader,
```

Comment:
Please rebase this after #447 is merged, it would be easier to review just the relevant changes here.